React/Modules/RCTAnimationDriverNode.m

```diff
@@ -0,0 +1,131 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "RCTAnimationDriverNode.h"
+#import <UIKit/UIKit.h>
+#import "RCTAnimation.h"
+
+const double SINGLE_FRAME_INTERVAL = 1.f / 60.f;
+
+@interface RCTAnimationDriverNode ()
+
+@property (nonatomic, copy) RCTResponseSenderBlock callback;
+
+@end
+
+@implementation RCTAnimationDriverNode {
+  NSArray *_frames;
+  double _delay;
+  double _toValue;
+  double _fromValue;
+  double _animationStartTime;
+  double _animationCurrentTime;
+  double _animationEndTime;
+  double _animationDuration;
+  RCTValueAnimatedNode *_valueNode;
+}
+
+- (instancetype)initWithId:(nonnull NSNumber *)animationId
+                     delay:(nullable NSNumber *)delay
+                   toValue:(nonnull NSNumber *)toValue
+                    frames:(nonnull NSArray *)frames
+                   forNode:(nonnull RCTValueAnimatedNode *)valueNode
+                  callBack:(nullable RCTResponseSenderBlock)callback {
+  self = [super init];
+  if (self) {
+    _animationId = animationId;
+    _toValue = toValue.doubleValue;
```

Comment:
@nicklockwood They are pulled from a dictionary when this method is called. So they have to be unboxed, I just unboxed them internally to the class instead of anywhere this method is called. I don't have a real preference here though.